packages/near-rpc-provider/lib/NearRpcProvider.ts

```diff
@@ -0,0 +1,185 @@
+import NodeProvider from '@liquality/node-provider'
+import { near, BigNumber, ChainProvider, FeeProvider } from '@liquality/types'
+import { NearNetwork } from '@liquality/near-networks'
+import { addressToString } from '@liquality/utils'
+import { normalizeTransactionObject, providers, Account } from '@liquality/near-utils'
+import { NodeError } from '@liquality/errors'
+
+import { get, isArray } from 'lodash'
+
+interface RpcProvider extends providers.JsonRpcProvider {
+  [key: string]: any
+}
+
+export default class NearRpcProvider extends NodeProvider implements Partial<ChainProvider>, FeeProvider {
+  _usedAddressCache: { [key: string]: boolean }
+  _accountsCache: { [key: string]: boolean }
+  _network: NearNetwork
+  _jsonRpc: RpcProvider
+
+  constructor(network: NearNetwork) {
+    super({
+      baseURL: network.helperUrl,
+      responseType: 'text',
+      transformResponse: undefined
+    })
+    this._jsonRpc = new providers.JsonRpcProvider(network.nodeUrl) as any
+    this._network = network
+    this._usedAddressCache = {}
+    this._accountsCache = {}
+  }
+
+  async sendRawTransaction(hash: string): Promise<string> {
+    const result = await this._sendRawTransaction(hash)
+    return get(result, 'transaction.hash')
+  }
+
+  async getBlockByHash(blockHash: string, includeTx: boolean) {
+    return this._getBlockById(blockHash, includeTx)
+  }
+
+  async getBlockByNumber(blockNumber: number, includeTx?: boolean) {
+    return this._getBlockById(blockNumber, includeTx)
+  }
+
+  async getBlockHeight(txHash?: string) {
+    const result = await this._rpc('block', txHash ? { blockId: txHash } : { finality: 'final' })
+    return get(result, 'header.height')
+  }
+
+  async getTransactionByHash(txHash: string): Promise<near.NormalizedTransaction> {
+    const currentHeight = await this.getBlockHeight()
+    const args = txHash.split('_')
+    const tx = await this._rpcQuery('tx', args)
+    const blockNumber = await this.getBlockHeight(tx.transaction_outcome.block_hash)
+    return normalizeTransactionObject({ ...tx, blockNumber }, currentHeight)
+  }
+
+  async getTransactionReceipt(txHash: string): Promise<near.InputTransaction> {
+    const args = txHash.split('_')
+    const tx = await this._rpcQuery('EXPERIMENTAL_tx_status', args)
+    const blockNumber = await this.getBlockHeight(tx.transaction_outcome.block_hash)
+    return { ...tx, blockNumber }
+  }
+
+  async getGasPrice() {
+    const result = await this._rpcQuery('gas_price', [null])
+    return get(result, 'gas_price')
+  }
+
+  async getBalance(addresses: string[]): Promise<BigNumber> {
```

Comment:
`getBalance` now takes `(addresses: (Address | string)[])`

Reply:
Should be fine now. I've also improved the way balance is calculated.